### ISSUE TITLE
Failed to create shader render pipeline, disabling shaders! 1.16.5

### ISSUE THREAD

**Issue by ImperfMage** [Asek3/Oculus#586]
### Minecraft Version

Minecraft 1.16.5

### Oculus Version

oculus-mc1.16.5-1.4.8

### Rubidium / Embeddium Version

embeddium-0.3.10+mc1.16.5

### Operating System

Windows 10

### What is your GPU?

Nvidia Rtx 3080ti

### Java Version

Java 8 

### Reproduction Steps

1. Open video settings 
2. Then open shaderpacks
3. Then open shader pack settings 
4. Open performance settings 
5. Turn on Texture filtering

### Crash Report file and latest.log

```shell
[23:37:16] [Render thread/ERROR] [Oculus/]: Failed to create shader rendering pipeline, disabling shaders!
java.lang.RuntimeException: Failed to create pass for gbuffers_terrain for rendering condition TERRAIN_OPAQUE specialized to input availability InputAvailability{texture=true, lightmap=true, overlay=false}
	at net.coderbot.iris.pipeline.DeferredWorldRenderingPipeline.lambda$new$2(DeferredWorldRenderingPipeline.java:344) ~[oculus:?] {re:classloading}
	at java.util.HashMap.computeIfAbsent(Unknown Source) ~[?:1.8.0_401] {}
	at net.coderbot.iris.pipeline.DeferredWorldRenderingPipeline.lambda$new$3(DeferredWorldRenderingPipeline.java:321) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.gbuffer_overrides.matching.ProgramTable.<init>(ProgramTable.java:16) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.pipeline.DeferredWorldRenderingPipeline.<init>(DeferredWorldRenderingPipeline.java:302) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.Iris.createPipeline(Iris.java:625) ~[oculus:?] {re:mixin,re:classloading}
	at net.coderbot.iris.pipeline.PipelineManager.preparePipeline(PipelineManager.java:37) ~[oculus:?] {re:mixin,re:classloading}
	at net.coderbot.iris.Iris.reload(Iris.java:574) ~[oculus:?] {re:mixin,re:classloading}
	at net.coderbot.iris.apiimpl.IrisApiV0ConfigImpl.setShadersEnabledAndApply(IrisApiV0ConfigImpl.java:28) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.gui.screen.ShaderPackScreen.applyChanges(ShaderPackScreen.java:526) ~[oculus:?] {re:classloading,re:mixin}
	at net.coderbot.iris.gui.screen.ShaderPackScreen.lambda$init$1(ShaderPackScreen.java:211) ~[oculus:?] {re:classloading,re:mixin}
	at net.minecraft.client.gui.widget.button.Button.func_230930_b_(SourceFile:33) ~[?:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.client.gui.widget.button.AbstractButton.func_230982_a_(SourceFile:16) ~[?:?] {re:classloading,re:mixin}
	at net.minecraft.client.gui.widget.Widget.func_231044_a_(Widget.java:136) ~[?:?] {re:classloading,pl:runtimedistcleaner:A,re:mixin,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.INestedGuiEventHandler.func_231044_a_(SourceFile:27) ~[?:?] {re:computing_frames,re:mixin,re:classloading}
	at net.minecraft.client.MouseHelper.func_198033_b(MouseHelper.java:87) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:mixins.betterthirdperson.json:MouseHelperMixin,pl:mixin:APP:magnesium_extras.mixins.json:Zoom.MouseMixin,pl:mixin:APP:mixins.betterthirdperson.json:MouseHelperFixupMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.Screen.func_231153_a_(Screen.java:427) ~[?:?] {re:computing_frames,pl:accesstransformer:B,xf:fml:configured:set_background_texture_2,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,xf:fml:configured:set_background_texture_2,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:configured:set_background_texture_2,pl:mixin:APP:legendarytooltips.mixins.json:ScreenMixin,pl:mixin:APP:quark.mixins.json:client.ScreenMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.MouseHelper.func_198023_a(MouseHelper.java:85) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:mixins.betterthirdperson.json:MouseHelperMixin,pl:mixin:APP:magnesium_extras.mixins.json:Zoom.MouseMixin,pl:mixin:APP:mixins.betterthirdperson.json:MouseHelperFixupMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.MouseHelper.func_228030_c_(MouseHelper.java:181) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:mixins.betterthirdperson.json:MouseHelperMixin,pl:mixin:APP:magnesium_extras.mixins.json:Zoom.MouseMixin,pl:mixin:APP:mixins.betterthirdperson.json:MouseHelperFixupMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.execute(SourceFile:94) ~[?:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.MouseHelper.func_228028_b_(MouseHelper.java:180) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:mixins.betterthirdperson.json:MouseHelperMixin,pl:mixin:APP:magnesium_extras.mixins.json:Zoom.MouseMixin,pl:mixin:APP:mixins.betterthirdperson.json:MouseHelperFixupMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36) ~[lwjgl-glfw-3.2.2.jar:build 10] {}
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.2.2.jar:build 10] {}
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3101) ~[lwjgl-glfw-3.2.2.jar:build 10] {}
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(SourceFile:109) ~[?:?] {re:mixin,re:classloading,pl:mixin:APP:mixins.oculus.json:MixinRenderSystem,pl:mixin:A}
	at net.minecraft.client.MainWindow.func_227802_e_(MainWindow.java:306) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:modernfix-forge.mixins.json:bugfix.preserve_early_window_pos.WindowMixin,pl:mixin:APP:magnesium_extras.mixins.json:BorderlessFullscreen.MainWindowAccessor,pl:mixin:APP:magnesium_extras.mixins.json:BorderlessFullscreen.WindowMixin,pl:mixin:APP:embeddiumextension.mixins.json:reduce_resolution_on_mac.MixinWindow,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:997) ~[?:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick_render_call,xf:fml:xaeroworldmap:xaero_wm_minecraftclient,xf:fml:xaerominimap:xaero_minecraftclient,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick_render_call,xf:fml:xaeroworldmap:xaero_wm_minecraftclient,xf:fml:xaerominimap:xaero_minecraftclient,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftMixin,pl:mixin:APP:modernfix-common.mixins.json:feature.measure_time.MinecraftMixin,pl:mixin:APP:modernfix-common.mixins.json:bugfix.concurrency.MinecraftMixin,pl:mixin:APP:modernfix-common.mixins.json:bugfix.world_leaks.MinecraftMixin,pl:mixin:APP:modernfix-forge.mixins.json:perf.skip_first_datapack_reload.MinecraftMixin,pl:mixin:APP:modernfix-forge.mixins.json:perf.blast_search_trees.MinecraftMixin,pl:mixin:APP:modernfix-forge.mixins.json:feature.measure_time.MinecraftMixin_Forge,pl:mixin:APP:mixins.oculus.json:MixinMinecraft_PipelineManagement,pl:mixin:APP:mixins.betterthirdperson.json:MinecraftMixin,pl:mixin:APP:magnesium_extras.mixins.json:FrameCounter.FpsAccessorMixin,pl:mixin:APP:memoryleakfix-16.mixins.json:targetEntityLeak.Minecraft_targetClearMixin,pl:mixin:APP:iceberg.mixins.json:MinecraftMixin,pl:mixin:APP:embeddium.mixins.json:features.options.MixinMinecraftClient,pl:mixin:APP:embeddium.mixins.json:features.gui.fast_fps_pie.MixinMinecraftClient,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) ~[?:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick_render_call,xf:fml:xaeroworldmap:xaero_wm_minecraftclient,xf:fml:xaerominimap:xaero_minecraftclient,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick,xf:fml:xaeroworldmap:xaero_wm_minecraft_runtick_render_call,xf:fml:xaeroworldmap:xaero_wm_minecraftclient,xf:fml:xaerominimap:xaero_minecraftclient,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftMixin,pl:mixin:APP:modernfix-common.mixins.json:feature.measure_time.MinecraftMixin,pl:mixin:APP:modernfix-common.mixins.json:bugfix.concurrency.MinecraftMixin,pl:mixin:APP:modernfix-common.mixins.json:bugfix.world_leaks.MinecraftMixin,pl:mixin:APP:modernfix-forge.mixins.json:perf.skip_first_datapack_reload.MinecraftMixin,pl:mixin:APP:modernfix-forge.mixins.json:perf.blast_search_trees.MinecraftMixin,pl:mixin:APP:modernfix-forge.mixins.json:feature.measure_time.MinecraftMixin_Forge,pl:mixin:APP:mixins.oculus.json:MixinMinecraft_PipelineManagement,pl:mixin:APP:mixins.betterthirdperson.json:MinecraftMixin,pl:mixin:APP:magnesium_extras.mixins.json:FrameCounter.FpsAccessorMixin,pl:mixin:APP:memoryleakfix-16.mixins.json:targetEntityLeak.Minecraft_targetClearMixin,pl:mixin:APP:iceberg.mixins.json:MinecraftMixin,pl:mixin:APP:embeddium.mixins.json:features.options.MixinMinecraftClient,pl:mixin:APP:embeddium.mixins.json:features.gui.fast_fps_pie.MixinMinecraftClient,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:184) ~[minecraft-1.16.5-client.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_401] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_401] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_401] {}
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_401] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:37) ~[forge-1.16.5-36.2.42-launcher.jar:36.2] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.1.3.jar:?] {re:classloading}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.1.3.jar:?] {re:classloading}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_401] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_401] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_401] {}
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_401] {}
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:67) [ForgeWrapper-prism-2024-02-29.jar:prism-2024-02-29] {}
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?] {}
Caused by: java.lang.RuntimeException: Failed to compile FRAGMENT shader for program gbuffers_terrain.fsh
	at net.coderbot.iris.gl.program.ProgramBuilder.buildShader(ProgramBuilder.java:101) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.gl.program.ProgramBuilder.begin(ProgramBuilder.java:52) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.pipeline.DeferredWorldRenderingPipeline.createPass(DeferredWorldRenderingPipeline.java:652) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.pipeline.DeferredWorldRenderingPipeline.lambda$new$2(DeferredWorldRenderingPipeline.java:342) ~[oculus:?] {re:classloading}
	... 46 more
Caused by: java.lang.RuntimeException: Shader compilation failed, see log for details
	at net.coderbot.iris.gl.shader.GlShader.createShader(GlShader.java:48) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.gl.shader.GlShader.<init>(GlShader.java:27) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.gl.program.ProgramBuilder.buildShader(ProgramBuilder.java:99) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.gl.program.ProgramBuilder.begin(ProgramBuilder.java:52) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.pipeline.DeferredWorldRenderingPipeline.createPass(DeferredWorldRenderingPipeline.java:652) ~[oculus:?] {re:classloading}
	at net.coderbot.iris.pipeline.DeferredWorldRenderingPipeline.lambda$new$2(DeferredWorldRenderingPipeline.java:342) ~[oculus:?] {re:classloading}
	... 46 more
```

### Additional context

Basically, once you try to turn on texture filtering it disables all of the shader. I don't know if this helps, but I'm running ComplementaryUnbound_r5.1.1

**Comment by ImperfMage** [comment]
Also made a new instance with just these mods with forge version 36.2.42
Embeddium [0.3.10+mc1.16.5]
ModernFix [5.14.0+mc1.16.5]
LazyDFU [0.1.3]
Memory Leak Fix [1.0.0]
TexTrue's Embeddium Options [0.1.2+mc1.16.5]
Smoothchunk mod [2.0]
Oculus 1.4.8
FerriteCore ((Neo)Forge) 2.1.1
Embeddium/Rubidium Extras v 1.4
and was able to reproduce the issue
